### PR TITLE
Tokenize correctly when assigning arrow function as object property

### PR DESCRIPTION
**Pull request by mnquintana**
Fixes https://github.com/atom/language-javascript/issues/221

/cc @50Wliu @atom/feedback

### CODE REVIEW

File: grammars/javascript.cson [hunk 1]
```diff
@@ -341,6 +341,31 @@
...
+      '3':
+        'name': 'keyword.operator.js'
+      '4':
+        'name': 'punctuation.definition.parameters.begin.js'
+    'end': '(\\))(\\s*=>)'
+    'endCaptures':
+      '1':
+        'name': 'punctuation.definition.parameters.end.js'
+      '2':
+        'name': 'storage.type.arrow.js'
```
Comment (MaximSokolov) [inline grammars/javascript.cson]: `storage.type.function.arrow` (see CoffeeScript)

Comment (50Wliu) [inline grammars/javascript.cson]: This should be addressed in a different PR (after this one is merged) to maintain consistency with the other rules.

File: spec/javascript-spec.coffee [hunk 2]
```diff
@@ -501,6 +501,15 @@ describe "Javascript grammar", ->
       expect(tokens[10]).toEqual value: ')', scopes: ['source.js', 'meta.function.arrow.js', 'punctuation.definition.parameters.end.js']
       expect(tokens[11]).toEqual value: '=>', scopes: ['source.js', 'meta.function.arrow.js', 'storage.type.arrow.js']
 
+    it "tokenizes arrow functions stored in object properties", ->
```
Comment (50Wliu) [inline spec/javascript-spec.coffee]: Any chance you can also test for `(param1, param2)`?

Comment (mnquintana) [inline spec/javascript-spec.coffee]: Sure - do you think that should be a separate spec or just modify this one?

**Comment by 50Wliu** [comment]
:articulated_lorry:
